src/main/java/me/lucien/minesweeper/domain/RoomManager.java

```diff
@@ -0,0 +1,113 @@
+package me.lucien.minesweeper.domain;
+
+import me.lucien.minesweeper.web.HttpException;
+import org.json.JSONObject;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+@Component
+public class RoomManager {
+
+    @Value("${room.capacity}")
+    private int capacity;
+
+    @Value("${room.delay}")
+    private int delay;
+
+    private Map<Integer, Room> roomMap = new HashMap<>();
+    private Map<Integer, ScheduledFuture> taskMap = new HashMap<>();
+    private ScheduledExecutorService executorService = Executors.newSingleThreadScheduledExecutor();
+
+    public String createRoom(int width, int height) throws HttpException {
+        if (roomMap.size() >= capacity) {
+            throw new HttpException(HttpStatus.SERVICE_UNAVAILABLE, "No available room.");
```

Comment:
Define a new exception or just return false. RoomManager should be independent of external APIs.